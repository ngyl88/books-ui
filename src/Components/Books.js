import React, { Component } from "react";
import { API_HOST } from "../config";

class Books extends Component {
  constructor() {
    super();
    this.state = {
      books: []
    };
  }
  async componentDidMount() {
    await this.getBooks();
  }
  render() {
    return (
      <div>
        <h3>Books</h3>
        {this.state.books.map(book => (
          <li key={book._id}>
            {book.title}, by {book.author ? book.author.name : ""}
          </li>
        ))}
      </div>
    );
  }
  async getBooks() {
    const url = `${API_HOST}/books`;
    const response = await fetch(url);
    if(response.ok) {
      const data = await response.json()
      this.setState({
        books: data
      });
    }
  }
}

export default Books;
