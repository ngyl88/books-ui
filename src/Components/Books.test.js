import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import Books from "./Books";

it("renders title", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Books />);

  const output = renderer.getRenderOutput();
  expect(output).toMatchSnapshot();
});

it("renders title and booklist after component mounts", async () => {
  fetch.mockResponseOnce(
    JSON.stringify([
      {
        _id: "10000001",
        title: "Harry Potter 1",
        author: {
          _id: "20000001",
          name: "author 1"
        }
      },
      {
        _id: "10000002",
        title: "Harry Potter 2"
      }
    ])
  );

  const renderer = new ShallowRenderer();
  renderer.render(<Books />);

  const instance = renderer.getMountedInstance();
  await instance.componentDidMount();

  const output = renderer.getRenderOutput();
  expect(output).toMatchSnapshot();
});
