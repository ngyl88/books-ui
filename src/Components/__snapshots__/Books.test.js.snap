// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders title 1`] = `
<div>
  <h3>
    Books
  </h3>
</div>
`;

exports[`renders title and booklist after component mounts 1`] = `
<div>
  <h3>
    Books
  </h3>
  <li>
    Harry Potter 1
    , by 
    author 1
  </li>
  <li>
    Harry Potter 2
    , by 
    
  </li>
</div>
`;
